budgeteer-web-interface/src/main/java/org/wickedsource/budgeteer/web/pages/contract/overview/table/TotalContractDetailsModel.java

```diff
@@ -0,0 +1,87 @@
+import static org.wicketstuff.lazymodel.LazyModel.model;
+
+public class TotalContractDetailsModel extends LoadableDetachableModel<ContractTotalData> implements IObjectClassAwareModel<ContractTotalData> {
+    private IModel<List<ContractTotalData>> wrappedModel;
+
+    public TotalContractDetailsModel(IModel<List<ContractBaseData>> sourceModel){
+        List<ContractTotalData> totalDataList = new ArrayList<>();
+
+        // Cast given sourceModel to a model of ContractTotalData.
+        for(ContractBaseData baseData: sourceModel.getObject())
+        {
+            totalDataList.add(new ContractTotalData(baseData));
+        }
+
+        wrappedModel = model(from(totalDataList));
+    }
+
+    /**
+     * Creates a list of emtpy strings.
+     * The length of the list is equal to the number of the contract attributes of a contract.
+     * @return a list of empty strings
+     */
+    public List<String> getContractAttributes()
+    {
+        List<DynamicAttributeField> fields = new ArrayList<DynamicAttributeField>();
+        if(wrappedModel.getObject().size()>0)
+        {
+            ContractTotalData data = wrappedModel.getObject().get(0);
+            fields = data.getContractAttributes();
+        }
+        List<String> result = new ArrayList<>();
+        for(DynamicAttributeField field: fields)
+        {
+            result.add("");
+        }
+        return result;
+    }
+
+    @Override
+    protected ContractTotalData load() {
+        ContractTotalData totalData = new ContractTotalData();
+
+        totalData.setBudgetLeft(MoneyUtil.createMoney(0d));
+        totalData.setBudget(MoneyUtil.createMoney(0d));
+        totalData.setBudgetSpent(MoneyUtil.createMoney(0d));
+
+        totalData.setBudgetGross(MoneyUtil.createMoney(0d));
+        totalData.setBudgetLeftGross(MoneyUtil.createMoney(0d));
+        totalData.setBudgetSpentGross(MoneyUtil.createMoney(0d));
+
+        // Sum up the money amounts for all contracts with and without taxes
+        for(ContractTotalData single : wrappedModel.getObject())
+        {
+            totalData.setBudgetSpent(totalData.getBudgetSpent().plus(single.getBudgetSpent()));
+            totalData.setBudget(totalData.getBudget().plus(single.getBudget()));
+            totalData.setBudgetLeft(totalData.getBudgetLeft().plus(single.getBudgetLeft()));
+
+            totalData.setBudgetGross(totalData.getBudgetGross().plus(MoneyUtil.getMoneyWithTaxes(single.getBudget(), BigDecimal.valueOf(single.getTaxRate()))));
+            totalData.setBudgetLeftGross(totalData.getBudgetLeftGross().plus(MoneyUtil.getMoneyWithTaxes(single.getBudgetLeft(), BigDecimal.valueOf(single.getTaxRate()))));
+            totalData.setBudgetSpentGross(totalData.getBudgetSpentGross().plus(MoneyUtil.getMoneyWithTaxes(single.getBudgetSpent(), BigDecimal.valueOf(single.getTaxRate()))));
+        }
+
+        return totalData;
+    }
+
+    public Class<ContractTotalData> getObjectClass() {
```

Comment:
If you don't implement the interface as I mentioned above then you also don't need this method.